public/app/features/dashboard/containers/NewDashboardWithDS.tsx

```diff
@@ -1,14 +1,19 @@
 import React, { useEffect, useState } from 'react';
 
-import { getDataSourceSrv, locationService } from '@grafana/runtime';
+import { config, getDataSourceSrv, locationService } from '@grafana/runtime';
 import { Page } from 'app/core/components/Page/Page';
 import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import { useLoadDataSource } from 'app/features/datasources/state';
+import { useDispatch } from 'app/types';
 
 import { getNewDashboardModelData, setDashboardToFetchFromLocalStorage } from '../state/initDashboard';
+import { setInitialDatasource } from '../state/reducers';
 
 export default function NewDashboardWithDS(props: GrafanaRouteComponentProps<{ datasourceUid: string }>) {
   const [error, setError] = useState<string | null>(null);
   const { datasourceUid } = props.match.params;
+  const dispatch = useDispatch();
+  useLoadDataSource(datasourceUid);
```

Comment:
Why do we need this?

Reply:
Oops, we don't. Leftover from earlier attempt.